### PR TITLE
Turn cleanup into boolean option, controlled by cli

### PR DESCRIPTION
**Pull request by ptnapoleon**
Now, cleanup will either be true or false. If true, it will use the previous tag functionality, but only with the tag of the current yaml.

### CODE REVIEW

**Comment by mambocab** [comment]
If I read this correctly, and if I understand your goals for our use of Jenkins, you might change this to a `--noclean` option that overrides `cleanup`. Then, we could pass `--noclean` to most of our calls, and this would be a less invasive change that could possibly be merged upstream.

If I'm wrong, then I need you to talk me through the problem again, and through the existing system's behavior. I don't really get the use of tags here.

**Comment by ptnapoleon** [comment]
My reasoning for requiring `cleanup` to need an arg, rather than `noclean` is as follows:
1. The vast majority of calls will not use cleanup, therefore it seems cleanup should be the special case.
2. Cleanup is a potentially destructive option, so it makes sense to explicitly require it

I do see how `noclean` would be more amenable to an upstream patch, but I'm not convinced that outweighs other considerations.

**Comment by ptnapoleon** [comment]
Tags exist per yaml and act as a sort of namespace. The existing functionality is that cleanup can exist in one of three states: true, false, or tag. If it is true, the running process will attempt to cleanup _any_ job created by autojobs. If false, it won't attempt any cleanup. If set to a tag, it will only search through the jobs who exist under the tag given. This is safest, because it prevents one yaml from touching jobs created by another.

In our set up, we will never need the existing true functionality. We will also never need to set it to a tag other than the current yaml tag (Hypothetically, one could have a dtest.yaml that uses the tag "dtest" but cleans up any with "utest" instead). So after this PR, if cleanup is false, the old false behavior is used. If cleanup is true or a string, we cleanup within the current tag.

File: jenkins_autojobs/main.py [hunk 1]
```diff
@@ -182,15 +183,11 @@ def cleanup(config, job_names, jenkins, verbose=True):
...
-                clean_tag = xml.xpath(tagxpath)
-                if not config['cleanup'] in clean_tag:
-                    continue
-            else:
-                print("Don't set cleanup to true, fool. You tried to delete all the jobs. For shame.")
-                raise Exception
+            # Only cleanup builds with the tag for this config.
+            xml = etree.fromstring(job.config.encode('utf8'))
+            clean_tag = xml.xpath(tagxpath)
+            if not config['tag'] in clean_tag:
```
Comment (mambocab) [inline jenkins_autojobs/main.py:189]: `clean_tag` should be a list, right?

**Comment by mambocab** [comment]
As long as that `in` check is performed on a list, not a string, I'm +1.

**Pull request merged**